selfdrive/modeld/parse_model_outputs.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+from typing import Dict
+from openpilot.selfdrive.modeld.constants import (
+  PLAN_MHP_N, PLAN_MHP_SELECTION, IDX_N, PLAN_WIDTH, NUM_LANE_LINES, LANE_LINES_WIDTH, NUM_ROAD_EDGES, POSE_WIDTH,
+  WIDE_FROM_DEVICE_WIDTH, LEAD_MHP_N, LEAD_MHP_SELECTION, LEAD_TRAJ_LEN, LEAD_WIDTH, DESIRE_PRED_WIDTH, DESIRE_PRED_LEN
+)
+
+def sigmoid(x):
+  return 1. / (1. + np.exp(-x))
+
+def softmax(x, axis=-1):
+  x -= np.max(x, axis=axis, keepdims=True)
+  if x.dtype == np.float32 or x.dtype == np.float64:
+    np.exp(x, out=x)
+  else:
+    x = np.exp(x)
+  x /= np.sum(x, axis=axis, keepdims=True)
+  return x
+
+def parse_mdn(name, outs, in_N=0, out_N=1, out_shape=None):
+  if name not in outs:
```

Comment:
why would this happen?

Reply:
It's useful for training, this parsing function is used for all driving sub-models, but not all of them output everything.

Reply:
made a `Parser` class with `ignore_missing` 
this also helps with reusing this code in training